zeroize/build.rs

```diff
@@ -13,42 +15,115 @@ fn main() {
+        /// Resolve the version of the installed C standard library
+        fn resolve() -> Self {
+            let output = Command::new("/usr/bin/ldd")
+                .arg("--version")
+                .output()
+                .unwrap();
 
-        if !output.status.success() {
-            panic!("/usr/bin/ldd --version exited with error: {:?}", output);
+            let stdout = String::from_utf8(output.stdout)
+                .unwrap()
+                .to_ascii_lowercase();
+            let stderr = String::from_utf8(output.stderr)
+                .unwrap()
+                .to_ascii_lowercase();
+
+            // Check if this is GNU C standard library
+            if stdout.find("glibc").is_some() || stderr.find("glibc").is_some() {
+                return Self::get_glibc_version();
+            }
+
+            // Check if this is musl-libc
+            if stdout.find("musl").is_some() || stderr.find("musl").is_some() {
+                return Self::get_musl_version();
+            }
+
+            StdLibrary::Unsupported
         }
 
-        let stdout = String::from_utf8(output.stdout).unwrap();
-        let info = stdout.split('\n').next().unwrap();
-        info.split(' ').last().unwrap().to_owned()
+        /// Get the version of the GNU C standard library
+        fn get_glibc_version() -> Self {
+            let output = Command::new("/usr/bin/ldd")
+                .arg("--version")
+                .output()
+                .unwrap();
+
+            if !output.status.success() {
+                panic!("/usr/bin/ldd --version exited with error: {:?}", output);
+            }
+
+            let stdout = String::from_utf8(output.stdout).unwrap();
+            let info = stdout.split('\n').next().unwrap();
+            let version =
+                Version::parse(&(info.split(' ').last().unwrap().to_owned() + ".0")).unwrap();
+
+            StdLibrary::GNU(version)
+        }
+
+        /// Get the version of the Musl C standard library
+        fn get_musl_version() -> Self {
+            let output = Command::new("/usr/bin/ldd")
+                .arg("--version")
+                .output()
+                .unwrap();
+
+            let stderr = String::from_utf8(output.stderr).unwrap();
+            let info = stderr.split('\n').collect::<Vec<&str>>()[1];
+            let version = Version::parse(info.split(' ').collect::<Vec<&str>>()[1]).unwrap();
+
+            StdLibrary::Musl(version)
+        }
+    }
+
+    /// Build `src/os/linux/explicit_bzero_backport.c` using the `cc` crate
+    pub fn build_explicit_bzero_backport() {
+        match StdLibrary::resolve() {
+            StdLibrary::GNU(ver) => {
+                if ver < Version::parse(GLIBC_WITH_EXPLICIT_BZERO).unwrap() {
+                    cc::Build::new()
+                        .file("src/os/linux/explicit_bzero_backport.c")
+                        .compile("explicit_bzero");
+                }
+            }
+
+            StdLibrary::Musl(ver) => {
```

Comment:
This looks nearly identical to the case above. Maybe you could move the `match` into a method on `StdLibrary` which decides whether to build `explicit_bzero_backport.c`, e.g. `needs_explicit_bzero_backport` returning a `bool` ?

Reply:
Yeah I was scratching my head there (late hours),  that's a good idea, will do.

Reply:
how about now?